### ISSUE TITLE
Native module cannot be null

### ISSUE THREAD

**Issue by bzytan** [fullstackreact/react-native-firestack#83]
Hi there, I followed the documented steps on how to do the manual linking on iOS.
After doing so I get  the attached result when attempting to import Firebase in my code with
import Firestack from 'react-native-firestack';

I'm wondering if I'm not doing the linking correctly or if its a result of incompatibility with my RN version (0.32). Is this something of a known issue?

![simulator screen shot 23 oct 2016 10 38 58 pm](https://cloud.githubusercontent.com/assets/2832933/19627098/fa0087ee-9971-11e6-8810-c9fc7bc89b12.png)

**Comment by auser** [comment]
Are you using cocoapods?

**Comment by moooji** [comment]
Yes, I am using cocoapods and I get the same error.

**Comment by auser** [comment]
Hm... what version of `react-native-firestack` are you running?

**Comment by moooji** [comment]
Ok, it is now working for me after a:
`````
rm rf node_modules/react-native-firestack
npm update react-native-firestack
rnpm link react-native-firestack
`````

I think the issue might have been that I used the git repo as package source in between. When I switched back to the published "2.3.2", it did not update my node_modules with the "official" version and kept the files from git.

**Comment by moooji** [comment]
I am using version 2.3.2 in a fresh RN 0.35 / 0.36 project using CocoaPods.

The installation guide says:

1. Install the npm package
2. Link the project with react-native link react-native-firestack
3. Modify the Build Phases for iOS like it says below.

I cannot find any information about 3. "Build Phases" though. Maybe that's the problem. What needs to be changed in Build Phases?

**Comment by auser** [comment]
Weird. I thought I took care of that already... I'll test over here. Thanks for the report. Also, I'll update the README. You no longer need to manipulate the build phases. :)

**Comment by Jeanmsilva89** [comment]
I followed the setup guide on a new project with RN @0.37.0 and I'm still seeing this issue. I also tried it on my current project that has RN 0.34.0 and had the same issue. I know this is closed but does anyone know the cause of this issue so I can try to figure out how to solve it?

**Comment by auser** [comment]
Yes, it means it can't find react-native. Meaning either the `react-native link` didn't work or the path to `react-native` didn't update in Firestack. Did you run `react-native link`?

**Comment by Jeanmsilva89** [comment]
I did use react native link

**Comment by auser** [comment]
Feel free to hop in the gittir channel for real-time chat. You can check it by looking at the `header search paths` in the fire stack project. Can you show me those dirs?

```bash
xcodebuild -showBuildSettings -project ./node_modules/react-native-firestack/ios/Firestack.xcodeproj | grep -i HEADER_SEARCH_PATHS
```

**Comment by Jeanmsilva89** [comment]
Cool I'll have to check that later tonight when I get home.. and I joined the gitter, will reach out to you later if I still need help.. Thanks :)

**Comment by florianbepunkt** [comment]
I had this issue several times myself. I think your installation instructions miss one step. What happens to me, when I do a fresh install is, that react-native link does create a podfile but it does not add the pods to it. So basically it links the library, but it passes with 0 pods installed. I need to do the following steps to make it work with rn 0.37.0. I hope this helps people that come across the issue. I am super new when it comes to xcode stuff, so wrote everything down so I remember it.


```
$ react-native init appName
$ cd appName
$ npm install react-native-firestack --save
```

create podfile in /ios (see content below make sure to change appName to name of your app). Just name the file Podfile (without extension, file name is case-sensitive).

```
$ react-native link react-native-firestack
$ cd ios && pod update --verbose
```

add ios app to firebase (following your installation instructions)

drag GoogleService-Info.plist to Xcode below appName (see screenshot), in prompt check "Copy items if needed", leave "added folders" at "Create groups" and make sure appName is checked in "Add to targets"

**Comment by leonzinger** [comment]
Hi Guys,

I am dealing with this problem for two days already. It still doesn't work.

I have a react-native v 0.40.0 firestack 2.3.5

followed all of the steps above but i still get this message "Native module cannot be null" after i add
`import Firestack from 'react-native-firestack'` in index.ios.js

**Comment by florianbepunkt** [comment]
@leonzinger Hi, I had this problem myself… using react-native-link on a fresh project it did NOT create the podfile (well it did, but an empty one).

so I needed to create a file called Podfile with the following content inside the `ios` folder

```
source 'https://github.com/CocoaPods/Specs.git'
platform :ios, '8.0'

def firestack_pods
  [
    'Firebase',
    'Firebase/Core',
    'Firebase/Auth',
    'Firebase/Storage',
    'Firebase/Database',
    'Firebase/RemoteConfig',
    'Firebase/Messaging'
  ].each do |lib|
    pod lib
  end
end

target 'YOURAPPNAMEHERE' do
  firestack_pods
end
```

after you created the podfile run

```
$ react-native link react-native-firestack
$ cd ios && pod update --verbose
```

**Comment by jvorcak** [comment]
For me it created a Podfile, but I still get the message `Native module cannot be null`

I have a react-native v 0.40.0 firestack 2.3.9, kind of the same as @leonzinger has.

**Comment by florianbepunkt** [comment]
@jvorcak can you upload a full repo (including node_modules and ios folder)?

**Comment by jvorcak** [comment]
@florianbepunkt I would, but unfortunately I can't :/ I tried to create a new project on a greenfield so that you can reproduce, but that one works. 

So hopefully the issue shouldn't pop up for other users and I'll just need to figure out why it fails in our project. Do you maybe have any hints where to look?

**Comment by florianbepunkt** [comment]
@jvorcak not really… hard to say without any code. but if you have a fresh project that works I would start going through the installation steps again, one by one and compare your project against the working one. this should help to pinpoint the issue. basically the error hints that the problem is not on the js side but somewhere in the linking process. so go through the manual steps for linking again

**Comment by auser** [comment]
I'm creating a fresh project right now... will report back on results shortly.

**Comment by auser** [comment]
Okay, so I created a full-blown new app with both Firestack and react-native-oauth from zero. It all works. There is some trickery around linking. 

* Does the app actually build?

https://gist.github.com/auser/9591a76f78234cb6586af4c1ef8eb92a

**Comment by jvorcak** [comment]
@auser as I stated above, it works for me for new app as well. The problem was that this existing application is older and we must have done something wrong. I personally think that this is a problem of our project. We had to solve it so we've created a new one and moved files there after trying so many things. I know it sucks, but it worked out.

The only think I'm thinking about is why @leonzinger experiences it as well...
